File: GithubUserJava/app/src/main/java/com/example/myapplication/repository/FavoriteRepository.java
package com.example.myapplication.repository;

import android.app.Application;
import androidx.lifecycle.LiveData;
import com.example.myapplication.data.local.entity.FavoriteEntity;
import com.example.myapplication.data.local.room.FavoriteDao;
import com.example.myapplication.data.local.room.FavoriteRoomDatabase;

import java.util.List;
import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

public class FavoriteRepository {

    private final FavoriteDao favDao;
    private final ExecutorService executorService = Executors.newSingleThreadExecutor();

    public FavoriteRepository(Application application) {
        FavoriteRoomDatabase db = FavoriteRoomDatabase.getDatabase(application);
        favDao = db.favDao();
    }

    public LiveData<List<FavoriteEntity>> getAllFavorites() {
        return favDao.getAllFavorite();
    }

    public LiveData<List<FavoriteEntity>> getUserFavoriteByUsername(String username) {
        return favDao.getUserFavoriteByUsername(username);
    }

    public void insert(FavoriteEntity fav) {
        executorService.execute(() -> favDao.insert(fav));
    }

    public void delete(FavoriteEntity fav) {
        executorService.execute(() -> favDao.delete(fav));
    }
}

File: GithubUserKotlin/app/src/main/java/com/example/myapplication/repository/FavoriteRepository.kt
package com.example.myapplication.repository

import android.app.Application
import androidx.lifecycle.LiveData
import com.example.myapplication.data.local.entity.FavoriteEntity
import com.example.myapplication.data.local.room.FavoriteDao
import com.example.myapplication.data.local.room.FavoriteRoomDatabase
import java.util.concurrent.ExecutorService
import java.util.concurrent.Executors

class FavoriteRepository(application: Application) {
    private val favDao: FavoriteDao
    private val executorService: ExecutorService = Executors.newSingleThreadExecutor()
    init {
        val db = FavoriteRoomDatabase.getDatabase(application)
        favDao = db.favDao()
    }
    fun getAllFavorites(): LiveData<List<FavoriteEntity>> = favDao.getAllFavorite()
    fun getUserFavoriteByUsername(username: String): LiveData<List<FavoriteEntity>> =
        favDao.getUserFavoriteByUsername(username)
    fun insert(fav: FavoriteEntity) {
        executorService.execute { favDao.insert(fav) }
    }
    fun delete(fav: FavoriteEntity) {
        executorService.execute { favDao.delete(fav) }
    }
}